src/mdast-process-images.js

```diff
@@ -44,17 +64,31 @@ export async function processImages(log, tree, mediaHandler, baseUrl) {
         node.url = new URL(url, baseUrl).href;
         register(node);
       } else if (url.startsWith('https://')) {
+        if (isExternalImage(url, externalImages)) {
+          // Add custom class to the node for external images
+          const data = node.data || {};
+          const hProperties = data.hProperties || {};
+          hProperties.className = 'external-image';
+          // eslint-disable-next-line no-param-reassign
+          node.data = { ...data, hProperties };
+
+          // eslint-disable-next-line no-param-reassign
+          node.data.externalImage = true;
+
+          log.debug(`Marked external image: ${url}`);
+        }
+
```

Comment:
why do you need to check it here and not just in `register()` ?

Reply:
Agreed. Simplified.